app/src/main/java/org/openobservatory/ooniprobe/common/MKCollectorResubmitTask.java

```diff
@@ -0,0 +1,87 @@
+package org.openobservatory.ooniprobe.common;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.WindowManager;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParser;
+
+import org.openobservatory.ooniprobe.R;
+import org.openobservatory.ooniprobe.dao.MeasurementDao;
+import org.openobservatory.ooniprobe.model.database.Measurement;
+
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+
+import androidx.appcompat.app.AppCompatActivity;
+import io.ooni.mk.MKCollectorResubmitResults;
+import io.ooni.mk.MKCollectorResubmitSettings;
+import localhost.toolkit.os.NetworkProgressAsyncTask;
+
+public class MKCollectorResubmitTask<A extends AppCompatActivity> extends NetworkProgressAsyncTask<A, Integer, Void> {
+	/**
+	 * {@code new MKCollectorResubmitTask(activity).execute(result_id, measurement_id);}
+	 *
+	 * @param activity from this task are executed
+	 */
+	public MKCollectorResubmitTask(A activity) {
+		super(activity, true, false);
+	}
+
+	private static void perform(Context c, Measurement m) throws IOException {
+		FileInputStream is = new FileInputStream(Measurement.getEntryFile(c, m.id, m.test_name));
+		String input = new GsonBuilder().disableHtmlEscaping().create().toJson(new JsonParser().parse(new InputStreamReader(is)));
```

Comment:
validazione e conversione a stringa

Reply:
The code that follows cannot avoid parsing the measurement because it needs to update the report ID.

For this reason I think we can skip this step here. If you don’t want to skip this step, then you should probably also validate the returned measurement to make sure that it is also JSON.

I do not have a strong opinion on what it is better to do. Maybe doing less work implies draining less battery. At the same time, more work implies more robustness (we used to have a bug where MK did not always emit valid JSON and this strategy would have helped to catch that).

I leave to you go choose the strategy you think it’s better here, just let’s be consistent.

Actually, another thing to keep in mind is whether we want to validate a measurement when it is emitted by MK rather than here. OTOH a measurement may be corrupted on disk, so from a conceptual and robustness oriented point of view, one should always validate what MK gives us.

Reply:
The code that follows cannot avoid parsing the measurement because it needs to update the report ID.

For this reason I think we can skip this step here. If you don’t want to skip this step, then you should probably also validate the returned measurement to make sure that it is also JSON.

I do not have a strong opinion on what it is better to do. Maybe doing less work implies draining less battery. At the same time, more work implies more robustness (we used to have a bug where MK did not always emit valid JSON and this strategy would have helped to catch that).

I leave to you go choose the strategy you think it’s better here, just let’s be consistent.

Actually, another thing to keep in mind is whether we want to validate a measurement when it is emitted by MK rather than here. OTOH a measurement may be corrupted on disk, so from a conceptual and robustness oriented point of view, one should always validate what MK gives us.

Reply:
actually i just checked file corruption

Reply:
Right, so I guess it also makes sense to verify what MK passes us, no?